src/StyledElement.js

```diff
@@ -79,53 +136,43 @@ class StyledElement extends BaseElement {
 	 * Adopt stylesheets
 	 */
 	adoptStyleSheets() {
-		if (!this.constructor['cssStyleSheets']) {
-			this.constructor['cssStyleSheets'] = this._styles.map((style) => {
-				const sheet = new CSSStyleSheet();
-				if (sheet && sheet.cssRules.length === 0) {
-					sheet.replaceSync(style);
-				}
-				return sheet;
-			});
-		}
+		const adoptGlobalStyleSheets = this._options.shadowRender && this._options.adoptGlobalStyles !== false;
 
-		if (StyledElement.globalStyles && !StyledElement['globalStyleSheet']) {
-			StyledElement['globalStyleSheet'] = new CSSStyleSheet();
-			StyledElement['globalStyleSheet'].replaceSync(StyledElement.globalStyles.textContent);
-		}
-
-		// adopt styles
-		// uses proposed solution for constructable stylesheets
-		// see: https://wicg.github.io/construct-stylesheets/#proposed-solution
 		this.getRoot().adoptedStyleSheets = [
-			...(this._options.shadowRender && this._options.adoptGlobalStyles && StyledElement['globalStyleSheet']
-				? [StyledElement['globalStyleSheet']]
-				: []),
-			...this.constructor['cssStyleSheets'],
+			...(adoptGlobalStyleSheets ? getGlobalStyleSheets(this._options.adoptGlobalStyles) : []),
+			...this.constructor['elementStyleSheets'],
 		];
 	}
 
 	/**
 	 * Custom polyfill for constructable stylesheets by appending styles to the end of an element
-	 * @param {string[]} styles
 	 */
-	appendStyleSheets(styles) {
+	appendStyleSheets() {
 		const parentDocument = getShadowParentOrBody(this.getRoot());
-		styles.forEach((style, index) => {
+
+		const adoptGlobalStyleSheets =
+			this._options.shadowRender && this._options.adoptGlobalStyles !== false && parentDocument !== document.body;
+
+		const appendableStyles = [
+			...(adoptGlobalStyleSheets ? getGlobalStyleSheets(this._options.adoptGlobalStyles) : []),
```

Comment:
@eddyloewen i actually don't think that a global mutation observer would be a bad idea, but as this i called in every components lifecycle the observer is called once per component.connect, right ?

And while thinking about this .. as this is a global feature .. do you think it could make sense to organize the global styles / observing in a dedicated GlobalStyleStore ?

Reply:
I doublechecked and i am pretty sure there are way to many observers instanciated even during lifecycle updates.

I created an alternative approach with a global Singleton Store which holds and indexes the sheets and provides a filter fn for the elements.

https://github.com/webtides/element-js/pull/105

Also still a Draft .. we should talk about this cause i don't understand it fully ...